### PR TITLE
Vehicles hear exterior audio

### PR DESCRIPTION
**Pull request by Doubleumc**
# About the pull request

As described in issue #4777 while vehicle interiors are intended to hear external sounds, it is largely nonfunctional. Sounds are either not heard or are inaudibly quiet. This is due to several smaller issues working together.

First, interiors are not detected as interiors. In commit https://github.com/cmss13-devs/cmss13/commit/0808500777f390bbe1d59142e955678344ef6d3e the types stored in the `used_turfs` list were changed to weakrefs, but the `is_interior()` function still checked for the old type, causing it to always return `FALSE`: https://github.com/cmss13-devs/cmss13/blob/5fbe14390f80225a5d295f07b9acf781fb8aa2b2/code/controllers/subsystem/interior.dm#L54-L58
Fixed by checking for the correct type.

Second, the `get_interior_by_coords()` function returned an object on success, but returned `FALSE` on failure - which still counted as an object. Since shuttles add turfs to `used_turfs` but don't have actual interiors, they would pass `is_interior()` but fail `get_interior_by_coords()`. The sound code expected any object received at this point to be valid, but using `FALSE` would thrown an error: https://github.com/cmss13-devs/cmss13/blob/5fbe14390f80225a5d295f07b9acf781fb8aa2b2/code/game/sound.dm#L82-L84
Fixed by returning `null` on failure.

Third, the audio was now playing but in the wrong part of the interior. It was offset to the top-right corner because `get_middle_coords()` got the difference between the bounds, but never actually averaged them: https://github.com/cmss13-devs/cmss13/blob/5fbe14390f80225a5d295f07b9acf781fb8aa2b2/code/modules/vehicles/interior/interior.dm#L321
Fixed by dividing the difference by two.

Fourth, the sound playback zone was oriented incorrectly because the width and height were transposed when creating the zone. Rectangles were defined as width then height, but the sound subsystem was passing height then width: https://github.com/cmss13-devs/cmss13/blob/5fbe14390f80225a5d295f07b9acf781fb8aa2b2/code/controllers/subsystem/sound.dm#L44
Fixed by passing width then height.

# Explain why it's good for the game

Vehicle interiors were intended to hear external sounds. This restores that functionality.
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
fix: vehicle interiors can hear exterior noises
/:cl:

### CODE REVIEW

**Comment by fira** [comment]
What do you mean by "FALSE is still counted as an object" ? FALSE is literally 0 which is falsey, this shouldn't change anything for the purposes of what's done here

**Comment by Doubleumc** [comment]
I might be using the wrong words here. What I mean is the sound code takes the return of `get_interior_by_coords()` (`VI`), checks if it exists (`VI?`), and uses the `ready` var from it (`VI?.ready`): https://github.com/cmss13-devs/cmss13/blob/5fbe14390f80225a5d295f07b9acf781fb8aa2b2/code/game/sound.dm#L83-L84

It doesn't check if its `TRUE` or `FALSE`, it checks if it exists or not. And `FALSE`/`0` exists, but can't be used in this way:
image

File: code/controllers/subsystem/interior.dm [hunk 1]
```diff
@@ -42,7 +42,7 @@ SUBSYSTEM_DEF(interior)
 			continue
 		if(x >= bounds[1].x && x <= bounds[2].x && y >= bounds[1].y && y <= bounds[2].y)
 			return current_interior
-	return FALSE
+	return null
```
Comment (harryob) [inline code/controllers/subsystem/interior.dm:45]:
```suggestion
	return
```

**Review by harryob — CHANGES_REQUESTED** [review]
works for me, thanks for the fix

**Review by harryob — APPROVED** [review]

**Pull request merged**